ui/models/perfSeries.js

```diff
@@ -0,0 +1,102 @@
+};
+
+export const getSeriesOptions = function (signatureProps, optionCollectionMap) {
+  let options = [optionCollectionMap[signatureProps.option_collection_hash]];
+  if (signatureProps.extra_options) {
+    options = options.concat(signatureProps.extra_options);
+  }
+  return [...new Set(options)];
+};
+
+export const getSeriesName = function (signatureProps, optionCollectionMap,
+                                displayOptions) {
+  const platform = signatureProps.machine_platform;
+  let name = getTestName(signatureProps);
+
+  if (displayOptions && displayOptions.includePlatformInName) {
+    name = `${name} ${platform}`;
+  }
+  const options = getSeriesOptions(signatureProps, optionCollectionMap);
+  return `${name} ${options.join(' ')}`;
+};
+
+export const getSeriesSummary = function (projectName, signature, signatureProps,
+                                   optionCollectionMap) {
+  const platform = signatureProps.machine_platform;
+  const options = getSeriesOptions(signatureProps, optionCollectionMap);
+
+  return {
+    id: signatureProps.id,
+    name: getSeriesName(signatureProps, optionCollectionMap),
+    testName: getTestName(signatureProps), // unadorned with platform/option info
+    suite: signatureProps.suite,
+    test: signatureProps.test || null,
+    signature: signature,
+    hasSubtests: signatureProps.has_subtests || false,
+    parentSignature: signatureProps.parent_signature || null,
+    projectName: projectName,
+    platform: platform,
+    options: options,
+    frameworkId: signatureProps.framework_id,
+    lowerIsBetter: (signatureProps.lower_is_better === undefined ||
+      signatureProps.lower_is_better),
+  };
+};
+
+export default class PerfSeriesModel {
+  static getSeriesList(projectName, params) {
+    return OptionCollectionModel.getMap().then(function (optionCollectionMap) {
+      return fetch(
+        `${getProjectUrl('/performance/signatures/', projectName)}${createApiQueryParams(params)}`,
+        ).then(async (resp) => {
+          if (resp.ok) {
+            const data = await resp.json();
+            return Object.entries(data).map(([signature, signatureProps]) => (
+              getSeriesSummary(projectName, signature, signatureProps, optionCollectionMap)
+            ));
+          }
+      });
+    });
+  }
+
+  static getPlatformList(projectName, params) {
+    return fetch(
+      `${getProjectUrl('/performance/platforms/', projectName)}${createApiQueryParams(params)}`,
+    ).then(async resp => resp.json());
+  }
+
+  static getSeriesData(projectName, params) {
+    return fetch(
+      `${getProjectUrl('/performance/data/', projectName)}${createApiQueryParams(params)}`,
+    ).then((resp) => {
+      if (resp.ok) {
+        return resp.json();
+      }
+      return Promise.reject('No series data found');
+    });
+  }
+
+  static getReplicateData(params) {
+    params.value = 'perfherder-data.json';
```

Comment:
Let's not modify the value of `params` after it's passed in, instead just use the spread operator to modify it when passing it to createApiQueryParams.

Reply:
Ahh, good point.  Fixed.